### ISSUE TITLE
Use the HEAD revision of MSL maintenance version instead of the official release

### ISSUE THREAD

**Issue by casella** [OpenModelica/OpenModelica#7966]
In [#6200 comment:19](https://trac.openmodelica.org/OpenModelica/ticket/6200#comment:19), @kabdelhak reports an issue with `Modelica.Mechanics.MultiBody.Examples.Constraints.PrismaticConstraint`, namely a singular initialization problem.

Following the discussion in MSL [#3875](https://github.com/modelica/ModelicaStandardLibrary/issues/3875), the issue has already been identified and fixed on the `maint/3.2.3` and `maint/4.0.x` branches.

It thought we were using the HEAD revisions of those two branches for the MSL, both for testing and for shipping with OpenModelica, so that we can benefit of the latest bugfixes, particularly the ones that affect our testing reports badly, without waiting for the official release. This is even more important now that the MSL release process is a bit stuck.

@adrpo, @sjoelund, could we just do that?

**Comment by casella** [comment]
@sjoelund, how is the package manager currently configured?

In fact, using maint/4.0.1 instead of 4.0.1 could be slightly problematic, because it has a version("4.0.1") annotation. No conversion required, but one may be puzzled why 4.0.1 is used if it hasn't been released yet...

**Comment by beutlich** [comment]
I'd claim the issue could be resolved upstream-wise if MAP-Lib sticks to its own [recommended release schedule](https://github.com/modelica/MAP-LIB_ProjectRules/blob/master/MAP-LIB_ProjectRules.md#annex-6-recommendation-for-a-release-process-for-libraries) with two releases per year, one (main) in March and one (maintenance) in September.

**Comment by casella** [comment]
I agree, but that is not currently in my power. Also, I think it would be nice to be able to see the positive influence of bugfixes, particularly those that fix models that are found to be invalid (e.g. because of language non-compliance, wrong initial equations, etc.), without a delay of up to 6 months.

**Comment by sjoelund** [comment]
> @sjoelund, how is the package manager currently configured?

It uses the HEAD revisions of branches. For example:
https://github.com/OpenModelica/OpenModelica-ModelicaStandardLibrary/tree/OM/maint%2F4.0.x
https://github.com/modelica/ModelicaStandardLibrary/tree/maint/4.0.x

**Comment by casella** [comment]
OK, maybe it's just @kabdelhak using the official release?

**Comment by sjoelund** [comment]
> OK, maybe it's just @kabdelhak using the official release?

Or maybe he installed 4.0.x at some point and didn't upgrade.

**Comment by casella** [comment]
I guess he was using MSL 3.2, as he wrote. No idea why, though.
